src/surface.cpp

```diff
@@ -978,6 +978,15 @@ SurfaceQuadric::distance(Position r, Direction ang, bool coincident) const
       d = (-k + sqrt(quad)) / a;
     }
 
+  } else if (a == 0.0) {
```

Comment:
I'm wondering if this case should come before the one above. In the case where a particle is coincident with a planar surface defined as a quadric, it looks like we'll enter the case above and end up dividing by zero still.

Reply:
Ah, true. I never hit this issue since I was only doing quadrics as problem boundaries, but I think non-boundary planar quadrics will give an error here as-is. Good catch!

Reply:
OK, the latest commit addresses this. I couldn't immediately come up with a more elegant way to write this.